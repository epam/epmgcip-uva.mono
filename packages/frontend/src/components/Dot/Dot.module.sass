@use 'src/style/colors'

.dot
   width: 4px
   height: 4px
   border-radius: 50%
   background-color: colors.$gray-200