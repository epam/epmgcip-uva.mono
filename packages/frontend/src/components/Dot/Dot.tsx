import css from './Dot.module.sass';

export const Dot = () => {
    return (<span className={css.dot}></span>)
}