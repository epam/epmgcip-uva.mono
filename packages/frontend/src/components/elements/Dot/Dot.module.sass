@use 'src/style/colors'

.dot
   width: 4px
   height: 4px
   border-radius: 50%

   &.green
      background-color: colors.$green
   &.gray
      background-color: colors.$gray-200
