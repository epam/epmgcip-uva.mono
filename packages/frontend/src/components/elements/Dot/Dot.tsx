import { getClassesList } from 'src/utils/getClassesList';
import css from './Dot.module.sass';

interface IDotProps {
  color?: 'gray' | 'green';
}

export const Dot = ({ color }: IDotProps) => {
  return <span className={getClassesList(css.dot, css[color ?? 'gray'])}></span>;
};
