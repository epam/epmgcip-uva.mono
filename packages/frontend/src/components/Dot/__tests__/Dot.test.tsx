import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { Dot } from '../Dot';

describe('Testing: Dot', () => {
  it('should render', () => {
    const { container } = render(<Dot />);

    expect(container).toMatchSnapshot();
  });
});
