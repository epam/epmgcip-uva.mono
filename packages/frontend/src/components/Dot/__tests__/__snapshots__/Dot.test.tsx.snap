// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: Dot > should render 1`] = `
<div>
  <span
    class="_dot_5e3be7"
  />
</div>
`;
